### PR TITLE
Additions to help use in Mesos+Marathon

### PR DESCRIPTION
**Pull request by misterbisson**
- Register in Consul with the Docker container's hostname
- Register a ttl health check for the container awaiting configuration

Together, those make it easier to find Couchbase instances that are awaiting configuration, then get info about them by container ID via the Docker API.

### CODE REVIEW

File: bin/triton-bootstrap [hunk 1]
```diff
@@ -205,11 +209,15 @@ else
     while [ $CLUSTERFOUND != 1 ]; do
         echo -n '.'
 
-        CLUSTERIP=$(curl -L -s -f curl http://consul:8500/v1/health/service/couchbase?passing | json -aH Service.Address | head -1)
```
Comment (misterbisson) [inline bin/triton-bootstrap:208]: stray `curl` removed

File: bin/triton-bootstrap [hunk 2]
```diff
@@ -149,7 +153,7 @@ then
     #
     # Deregister this instance from the list of unconfigured instances in Consul
     #
-    curl -f --retry 7 --retry-delay 3 http://consul:8500/v1/agent/service/deregister/couchbase-unconfigured-$MYIPPRIVATE
+    curl -f --retry 7 --retry-delay 3 http://consul:8500/v1/agent/service/deregister/couchbase-unconfigured-$HOSTNAME
```
Comment (misterbisson) [inline bin/triton-bootstrap:156]: `$HOSTNAME` is the container ID

File: bin/triton-bootstrap [hunk 3]
```diff
@@ -92,7 +92,11 @@ fi
 #
 # Register this unconfigured Couchbase instance in Consul for discovery by the configuration/bootstrap agent
 #
-curl -f --retry 7 --retry-delay 3 http://consul:8500/v1/agent/service/register -d "$(printf '{"ID":"couchbase-unconfigured-%s","Name":"couchbase-unconfigured","Address":"%s"}' $MYIPPRIVATE $MYIPPRIVATE)"
+curl -f --retry 7 --retry-delay 3 http://consul:8500/v1/agent/service/register -d "$(printf '{"ID":"couchbase-unconfigured-%s","Name":"couchbase-unconfigured","Address":"%s","checks": [{"ttl": "5900s"}]}' $HOSTNAME $MYIPPRIVATE)"
+
+# pass the healthcheck
+curl -f --retry 7 --retry-delay 3 "http://consul:8500/v1/agent/check/pass/service:couchbase-unconfigured-$HOSTNAME?note=initial+startup"
```
Comment (misterbisson) [inline bin/triton-bootstrap:98]: The TTL health check fails until we explicitly pass it.

File: bin/triton-bootstrap [hunk 4]
```diff
@@ -92,7 +92,11 @@ fi
 #
 # Register this unconfigured Couchbase instance in Consul for discovery by the configuration/bootstrap agent
 #
-curl -f --retry 7 --retry-delay 3 http://consul:8500/v1/agent/service/register -d "$(printf '{"ID":"couchbase-unconfigured-%s","Name":"couchbase-unconfigured","Address":"%s"}' $MYIPPRIVATE $MYIPPRIVATE)"
+curl -f --retry 7 --retry-delay 3 http://consul:8500/v1/agent/service/register -d "$(printf '{"ID":"couchbase-unconfigured-%s","Name":"couchbase-unconfigured","Address":"%s","checks": [{"ttl": "5900s"}]}' $HOSTNAME $MYIPPRIVATE)"
```
Comment (misterbisson) [inline bin/triton-bootstrap:95]: Register this Couchbase instance as a service that's awaiting configuration. It gets de-registered later, or will fail out after the TTL health check expires.

**Comment by xer0x** [comment]
lgtm. The TTL timer counts down inside Consul? I'll have to read up on healthchecks.

**Comment by misterbisson** [comment]
> The TTL timer counts down inside Consul

Right. It's the bomb detonation timer since the last time it received a `/pass` hit.

**Pull request merged**